pymeasure/instruments/rohdeschwarz/fsseries.py

```diff
@@ -0,0 +1,409 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+import warnings
+
+import numpy as np
+
+from pymeasure.instruments import Instrument, SCPIMixin
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def _number_or_auto(value):
+    # helper for the bandwidth setting
+    if isinstance(value, str) and value.upper() == "AUTO":
+        return ":AUTO ON"
+    else:
+        # There is no space in the set commands, so we have to add it
+        return " " + str(value)
+
+
+class FSSeries(SCPIMixin, Instrument):
+    """
+    Represents a Rohde&Schwarz FS Series of spectrum analyzers like FSL and FSW.
+
+    All physical values that can be set can either be as a string of a value and a unit (e.g.
+    "1.2 GHz") or as a float value in the base units (Hz, dBm, etc.).
+    """
+
+    def __init__(self, adapter, name="Rohde&Schwarz FS series instrument", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    # Frequency settings ---------------------------------------------------------------------------
+
+    freq_span = Instrument.control(
+        "FREQ:SPAN?",
+        "FREQ:SPAN %s",
+        "Control frequency span in Hz.",
+    )
+
+    freq_center = Instrument.control(
+        "FREQ:CENT?",
+        "FREQ:CENT %s",
+        "Control center frequency in Hz.",
+    )
+
+    freq_start = Instrument.control(
+        "FREQ:STAR?",
+        "FREQ:STAR %s",
+        "Control start frequency in Hz.",
+    )
+
+    freq_stop = Instrument.control(
+        "FREQ:STOP?",
+        "FREQ:STOP %s",
+        "Control stop frequency in Hz.",
+    )
+
+    attenuation = Instrument.control(
+        "INP:ATT?",
+        "INP:ATT %s",
+        "Control attenuation in dB.",
+    )
+
+    res_bandwidth = Instrument.control(
+        "BAND:RES?",
+        # There is no space between RES and %s on purpose, see _number_or_auto.
+        "BAND:RES%s",
+        "Control resolution bandwidth in Hz. Can be set to 'AUTO'",
+        set_process=_number_or_auto,
+    )
+
+    video_bandwidth = Instrument.control(
+        "BAND:VID?",
+        "BAND:VID%s",
+        "Control video bandwidth in Hz. Can be set to 'AUTO'",
+        set_process=_number_or_auto,
+    )
+
+    # Sweeping -------------------------------------------------------------------------------------
+
+    sweep_time = Instrument.control(
+        "SWE:TIME?",
+        # No space between TIME and %s on purpose, see _number_or_auto.
+        "SWE:TIME%s",
+        "Control sweep time in s. Can be set to 'AUTO'.",
+        set_process=_number_or_auto,
+    )
+
+    continuous_sweep = Instrument.control(
```

Comment:
For boolean properties, names like `continuous_sweep_enabled`  are good, as they make it understandable, what True and False mean.